tests/integration/test_integration.py

```diff
@@ -124,20 +150,27 @@ async def test_given_certificates_provider_is_related_when_vault_status_checked_
 ):
     """To test that Vault is actually running when the charm is active."""
     assert ops_test.model
-    await ops_test.model.wait_for_idle(
-        apps=[APP_NAME],
-        status="active",
-        timeout=1000,
-    )
+    async with ops_test.fast_forward():
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME],
+            status="blocked",
+            timeout=1000,
+        )
     await ops_test.model.integrate(
         relation1=f"{SELF_SIGNED_CERTIFICATES_APPLICATION_NAME}:certificates",
         relation2=f"{APP_NAME}:tls-certificates-access",
     )
-    await ops_test.model.wait_for_idle(
-        apps=[APP_NAME, SELF_SIGNED_CERTIFICATES_APPLICATION_NAME],
-        status="active",
-        timeout=1000,
-    )
+    async with ops_test.fast_forward():
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME],
+            status="blocked",
+            timeout=1000,
+        )
+        await ops_test.model.wait_for_idle(
+            apps=[SELF_SIGNED_CERTIFICATES_APPLICATION_NAME],
+            status="active",
+            timeout=1000,
+        )
     unit_ip = ops_test.model.units.get(f"{APP_NAME}/0").public_address
     vault_endpoint = f"https://{unit_ip}:8200"
     action_output = await run_get_ca_certificate_action(ops_test)
```

Comment:
Is this working for you? as far as I know the vault cert isn't signed for the unit IP, just the application ip and socket fqdn.

Reply:
The machine charm is slightly different in this respect, and the IP address is actually used. This didn't work on k8s due to how pods are effectively disposable, but seems to be reasonable on the machine charm.